lib/opentelemetry/trace/exporter/otlp.lua

```diff
@@ -96,12 +96,59 @@ function _M.export_spans(self, spans)
             }
         }
     }
-    for _, span in ipairs(spans) do
+    local tracers = {}
+    local providers = {}
+    tracers[spans[1].tracer] = 1
+    providers[spans[1].tracer.provider] = 1
+    for i, span in ipairs(spans) do
+        local rs_idx = providers[span.tracer.provider]
+        local ils_idx = tracers[span.tracer]
+        if rs_idx == nil then
+            rs_idx = #body.resource_spans + 1
+            ils_idx = 1
+            providers[span.tracer.provider] = rs_idx
+            tracers[span.tracer] = ils_idx
+            table.insert(
+                body.resource_spans,
+                {
+                    resource = {
+                        attributes = spans[i].tracer.provider.resource.attrs,
```

Comment:
why use `spans[i]` here instead of `span`?

Reply:
legacy holdover of a previous approach where i was using `i` more frequently, I'll remove